File: frontend/src/types/role.ts
// ============================================================
// Company OS — Role & Permission Types
// ============================================================

export interface Permission {
  id: string;
  code: string;
  module: string;
  description?: string;
  created_at: string;
}

export interface Role {
  id: string;
  name: string;
  code: string;
  description?: string;
  is_system: boolean;
  created_at: string;
  updated_at: string;
  permissions: Permission[];
}

export interface RoleCreate {
  name: string;
  code: string;
  description?: string;
}

export interface RoleUpdate {
  name?: string;
  description?: string;
}

export interface RolePermissionUpdate {
  permission_ids: string[];
}

File: app/schemas/role.py
import uuid
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class PermissionOut(BaseModel):
    id: uuid.UUID
    code: str
    module: str
    description: Optional[str] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class RoleBase(BaseModel):
    name: str
    code: str
    description: Optional[str] = None


class RoleCreate(RoleBase):
    pass


class RoleUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class RoleOut(RoleBase):
    id: uuid.UUID
    is_system: bool
    created_at: datetime
    updated_at: datetime
    permissions: List[PermissionOut] = []

    model_config = ConfigDict(from_attributes=True)


class RolePermissionUpdate(BaseModel):
    permission_ids: List[uuid.UUID]
